wire-ios-data-model/Tests/ConversationSearch/FilterConversationsUseCaseTests.swift

```diff
@@ -0,0 +1,205 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireDataModelSupport
+import XCTest
+
+@testable import WireDataModel
+
+final class FilterConversationsUseCaseTests: XCTestCase {
+
+    private var sut: FilterConversationsUseCase<MockContainer>!
+
+    override func tearDown() {
+        sut = nil
+    }
+
+    func testEmptySearchTextReturnsEverything() {
+
+        // Given
+        sut = .init(conversationContainers: conversationContainers)
+
+        // When
+        let filtered = sut.invoke(query: "")
+
+        // Then
+        XCTAssertEqual(filtered, conversationContainers)
+    }
+
+    func testSingleCharacterSearchTextReturnsNoConversations() {
+
+        // Given
+        sut = .init(conversationContainers: conversationContainers)
+
+        // When
+        let filtered = sut.invoke(query: "y")
```

Comment:
I don't think the data contains a "y". If you change this to an "a" I think you test will fail right? Maybe I misunderstand the purpose of this test. I thought it means that any single character search will always return no conversations

Reply:
The name was misleading. The test was about not matching anything.
I changed the name.